subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformBuildOperationIntegrationTest.groovy

```diff
@@ -896,10 +897,19 @@ class ArtifactTransformBuildOperationIntegrationTest extends AbstractIntegration
     }
 
     def getPlannedNodes(int transformNodeCount) {
-        def plannedNodes = buildOperations.only(CalculateTaskGraphBuildOperationType).result.executionPlan as List<PlannedNode>
-        assert plannedNodes.every { KNOWN_NODE_TYPES.contains(it.nodeIdentity.nodeType) }
-        assert plannedNodes.count { it.nodeIdentity.nodeType.toString() == TRANSFORM_STEP } == transformNodeCount
-        return plannedNodes
+        def ops = buildOperations.all(CalculateTaskGraphBuildOperationType)
+        assert !ops.empty
+        if (GradleContextualExecuter.configCache) {
+            assert ops.size() <= 2
```

Comment:
❔ Why is it `<= 2` and not always `2`?

Reply:
Because we currently don't run the test with load after store. I added a check so there is a failure prompting you to changed this to `==`.